### ISSUE TITLE
Implement Tab Groups settings on Android

### ISSUE THREAD

**Issue by heapifyman** [brave/brave-browser#31289]
## Description 
As described in [How do I Enable/Disable Tab Groups on Android?](https://support.brave.com/hc/en-us/articles/4402426130573-How-do-I-Enable-Disable-Tab-Groups-on-Android-) it should be possible to completely disable Tab Groups.

But the setting "Enable Tab Groups" does not exist any more in version `1.52.126`.

## Steps to reproduce 

   1. In brave://flags enable `Android tab groups settings`
   2. Make sure new settings in `Tabs and tab groups` appear and function as expected
   3. Expected behaviour is defined by sub-issues below and may be check on issue by issue basis:
       https://github.com/brave/brave-browser/issues/56928
       https://github.com/brave/brave-browser/issues/57253
       https://github.com/brave/brave-browser/issues/57491
       https://github.com/brave/brave-browser/issues/57524
       https://github.com/brave/brave-browser/issues/57618
       https://github.com/brave/brave-browser/issues/57732
       https://github.com/brave/brave-browser/issues/57837
       https://github.com/brave/brave-browser/issues/58014

## Actual result 

No option to disable Tab Groups

## Expected result

An option to disable Tab Groups.

It should be possible to completely disable the Tab Groups feature, so that there is no option to "Open in new tab in group any more when long-pressing a link - (or at least it should not be top-most option, see #30813 ).

## Issue reproduces how often 

Easily reproduced

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? Yes
- Can you reproduce this issue with the current Play Store Beta version? Yes
- Can you reproduce this issue with the current Play Store Nightly version? Yes

## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): Phone
   - Android version: 13, security update June 2023
   
## Brave version
`1.52.126`

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? no
- Does the issue resolve itself when disabling Brave Rewards? no
- Is the issue reproducible on the latest version of Chrome? Google Chrome does not have any settings to configure tab groups.

### Additional information

#30159 mentions that the option has moved to `Settings -> Accessibility -> Semplified view open tab` but

- this setting does not exist any more, either
- there is only "Simplified view for web pages (Get notified when an article can be shown in simplified view)", which is not related to disabling Tab Groups

**Comment by chengnan049** [comment]
Same problem in Brave Beta 1.54.76

**Comment by pterozactyl** [comment]
Still a problem. I've been patient but I guess they don't care

**Comment by defactl** [comment]
Has anyone found a solution to this problem?

**Comment by rafael-faria** [comment]
Still waiting on this feature. Groups seem to be getting worse, having the option to completely disable them would help

**Comment by addohm** [comment]
I'd like to see a way to disable them as well as many others. I have absolutely no need for tab groups on mobile and even still don't use them on desktop. All it's doing is creating extra clicks. At the very least you could create an option that disables opening links into a new tab group.

https://community.brave.com/t/disable-open-new-tab-in-group-option/490691

https://community.brave.com/t/turn-off-tab-groups/570997

With the current tab group implementation and also the forward/back button press-and-hold for history removal I can no longer support brave unfortunately. It's a very sad day when the best browsers out there come from data leeching privacy breaching POS corporations. Sorry for the rant. Looking forward to using brave again when this stuff gets worked out. The sync chain feature is pure genius.

**Comment by bsclifton** [comment]
cc: @timchilds for prioritization

**Comment by timchilds** [comment]
@samartnik for after large screen fixes. Remove from long press menu and other menus only. No need to remove drag/drop tab group creation.

**Comment by timchilds** [comment]
Location: Settings > Tabs and tab groups

Title: Tab groups
Subtitle: Enables tab groups
Default state: enabled

cc @aguscruiz

**Comment by aguscruiz** [comment]
Figma: 
https://www.figma.com/design/H11ZOl6JMYbCTW4ZJXqR5V/%F0%9F%A6%81-Browser?node-id=12863-30057&t=dq31szZ8Xr9qNUTy-1

<img width="824" height="1834" alt="Image" src="https://github.com/user-attachments/assets/b1a2eee3-93fa-4daf-8bff-99a98fcb552a" />

**Comment by zjawkmka** [comment]
Yes please make it gone or I'm forced to look for a new browser.

**Comment by samartnik** [comment]
Closing as the last sub-issue has been merged
Released (behind the flag) in v1.95.57

**Comment by Uni-verse** [comment]
Verified on `Pixel 9a` using the following version: 

```
Brave	1.95.100 Chromium: 153.0.8010.37 (Official Build) (64-bit) 
Revision	3d751bdf5d610fb427af6007ae17bd7b44f90bcd
OS	Android 16; Build/CP1A.260505.005; 36; REL
```
- Ensured that enabling `brave://flags#brave-android-tab-groups-settings` will display the `Groups` section in the `Tabs and tab groups` settings page.
- Ensured that `Enable tab groups` preference switch is enabled by default if the flag enabled.
- Ensured that  `New tab group` option is only shown in the tab switcher menu when feature is enabled. encountered: https://github.com/brave/brave-browser/issues/58880
- Ensured that `Tab groups bar` preference is respected and shows the tab groups tab when grouped.
- Ensured that opening links will automatically open in the same tab group.
- Ensured that disabling `Enable tab groups` toggle will display a confirmation dialog `Any existing groups will be deleted...tabs within those groups will become open, ungrouped...` will `Cancel` and `Disable`
- Ensured that grouped tabs will become ungrouped when disabling tab groups.

example | example | example | example
--|--|--|--
<img width="655" height="1360" alt="Image" src="https://github.com/user-attachments/assets/3b637fed-d5a9-409a-845a-83eb76c625d5" /> | <img width="655" height="1360" alt="Image" src="https://github.com/user-attachments/assets/052c4eac-6db2-407f-b259-3801900058f6" /> | <img width="655" height="1360" alt="Image" src="https://github.com/user-attachments/assets/89750557-f12c-4508-8008-0130e120f493" /> | <img width="655" height="1360" alt="Image" src="https://github.com/user-attachments/assets/06ebe21e-164e-4ad0-a7e5-52d18cf621cb" />
<img width="655" height="1360" alt="Image" src="https://github.com/user-attachments/assets/08abda25-e3ad-4e4b-8caf-f6da84933286" /> | <img width="655" height="1360" alt="Image" src="https://github.com/user-attachments/assets/e33bb9c5-262c-447a-bdd2-fe794eec9f29" /> | <img width="655" height="1360" alt="Image" src="https://github.com/user-attachments/assets/a46e5b15-077a-45b5-a585-f280216dea14" /> | <img width="655" height="1360" alt="Image" src="https://github.com/user-attachments/assets/ff0cc8b2-4d8f-4188-914d-2569fe19fead" />
